providers/implementations/rands/seeding/rand_unix.c

```diff
@@ -37,6 +37,24 @@
 # include <sys/param.h>
 #endif
 
+/*
+ * Provide a compile time error if the FIPS module is being build and none
+ * of the supported entropy sources are available.
+ */
+#if defined(FIPS_MODULE)
+# if !defined(OPENSSL_RAND_SEED_GETRANDOM) \
+     && !defined(OPENSSL_RAND_SEED_DEVRANDOM) \
+     && !defined(OPENSSL_RAND_SEED_RDCPU) \
+     && !defined(OPENSSL_RAND_SEED_OS)
+#  error FIPS mode without supported randomness source
+# endif
+/* Remove the sources that are not permitted in FIPS */
+# undef OPENSSL_RAND_SEED_LIBRANDOM
```

Comment:
How about adding a `#warning` for each if it is defined?

Reply:
This would break warnings as errors builds where the sources are wanted outside of FIPS mode.

Reply:
Okay.